sql/src/main/java/io/crate/executor/transport/TransportExecutor.java

```diff
@@ -321,29 +322,36 @@ private ExecutionPhasesTask executionPhasesTask(Plan plan, Job job) {
         }
     }
 
-    static class BulkNodeOperationTreeGenerator extends PlanVisitor<List<NodeOperationTree>, Void> {
+    static class BulkNodeOperationTreeGenerator extends PlanVisitor<BulkNodeOperationTreeGenerator.Context, Void> {
 
         NodeOperationTreeGenerator nodeOperationTreeGenerator = new NodeOperationTreeGenerator();
 
-        public List<NodeOperationTree> createNodeOperationTrees(Plan plan) {
-            ArrayList<NodeOperationTree> nodeOperationTrees = new ArrayList<>();
-            process(plan, nodeOperationTrees);
-            return nodeOperationTrees;
+        public List<NodeOperationTree> createNodeOperationTrees(Plan plan, String localNodeId) {
+            Context context = new Context();
+            context.nodeOperationTrees = new ArrayList<>();
```

Comment:
can you change the context ctor to take the localNodeId and also initialize the arrayList in the ctor?